### ISSUE TITLE
Flavors: Add `markform` flavor support for Markform `.form.md` files

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#404]
## Problem

rumdl flavors help lint Markdown variants without false positives. Markform documents are valid Markdown, but they rely on Markform-specific structure that `standard` rules commonly flag, especially:
- Required YAML front matter at the top of the file.
- Structural tags expressed as HTML comments (for example `<!-- form ... -->`, `<!-- field ... -->`).

## Request

Add a new flavor: `markform`, based on `standard`, aligned with the Markform MF/0.1 spec.

```toml
[global]
flavor = "markform"

[per-file-flavor]
"**/*.form.md" = "markform"
```

### Auto-detection

When no flavor is configured, auto-detect:
- `*.form.md` -> `markform`

## Syntax to recognize

Treat these as first-class Markform constructs so they do not trigger spurious violations:

### YAML front matter (required)

```yaml
---
markform:
  spec: MF/0.1
  title: Example
---
```

### Structural HTML comment tags

```text
<!-- form id="example" -->
<!-- group id="inputs" -->
<!-- field kind="string" id="name" label="Name" required=true --><!-- /field -->
<!-- /group -->
<!-- /form -->
```

## Expected rule behavior

- `MD041`: ignore Markform YAML front matter (and any leading structural comment tags) when determining the first heading.
- `MD033`: allow Markform structural HTML comments (at minimum `<!-- ... -->`) in `markform` flavor.

## Acceptance criteria

- `markform` is a supported `flavor` value and documented in `docs/flavors.md`.
- `*.form.md` is auto-detected as `markform` when no flavor is set.
- Add fixtures/tests demonstrating reduced false positives for `MD041` and `MD033` on valid Markform files.

## References

- https://github.com/jlevy/markform/blob/main/README.md
- https://github.com/jlevy/markform/blob/main/docs/markform-spec.md
- https://github.com/jlevy/markform/blob/main/docs/markform-reference.md

**Comment by rvben** [comment]
I see https://github.com/jlevy/markform has 41 stars, so I am not sure there's enough traction yet to start supporting it.
